msal/tests/Test.MSAL.NET.Integration/UnifiedCacheTests.cs

```diff
@@ -0,0 +1,491 @@
+        private adal::Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationResult adalAuthResult;
+        private void InitAdal()
+        {
+            adalCache = new adal::Microsoft.IdentityModel.Clients.ActiveDirectory.TokenCache()
+            {
+                BeforeAccess = AdalDoBefore,
+                AfterAccess = AdalDoAfter
+            };
+            adalContext = new adal::Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext(Authority, adalCache);
+        }
+
+        private msal::Microsoft.Identity.Client.PublicClientApplication msalPublicClient;
+        private msal::Microsoft.Identity.Client.TokenCache msalCache;
+        private msal::Microsoft.Identity.Client.AuthenticationResult msalAuthResult;
+        private void InitMsal()
+        {
+            msalCache = new msal::Microsoft.Identity.Client.TokenCache();
+
+            msal::Microsoft.Identity.Client.TokenCacheExtensions.SetBeforeAccess(msalCache, MsalDoBefore);
+            msal::Microsoft.Identity.Client.TokenCacheExtensions.SetAfterAccess(msalCache, MsalDoAfter);
+
+            msalPublicClient = new msal::Microsoft.Identity.Client.PublicClientApplication(ClientId, Authority, msalCache);
+        }
+
+        private void ValidateMsalAuthResult()
+        {
+            Assert.IsNotNull(msalAuthResult);
+            Assert.IsNotNull(msalAuthResult.AccessToken);
+            Assert.IsNotNull(msalAuthResult.IdToken);
+            Assert.AreEqual(user.Upn, msalAuthResult.Account.Username);
+        }
+
+        private void ValidateAdalAuthResult()
+        {
+            Assert.IsNotNull(adalAuthResult);
+            Assert.IsNotNull(adalAuthResult.AccessToken);
+            Assert.IsNotNull(adalAuthResult.IdToken);
+            Assert.AreEqual(user.Upn, adalAuthResult.UserInfo.DisplayableId);
+        }
+
+        [TestMethod]
+        [TestCategory("UnifiedCache_IntegrationTests")]
+        public async Task UnifiedCache_Msal2_Adalv4_SharedRtGetUsed()
+        {
+            msalAuthResult =
+                await msalPublicClient.AcquireTokenByUsernamePasswordAsync(MsalScopes, user.Upn, securePassword);
+            ValidateMsalAuthResult();
+
+            msalAuthResult =
+                await msalPublicClient.AcquireTokenByUsernamePasswordAsync(MsalScopes1, user.Upn, securePassword);
+            ValidateMsalAuthResult();
+
+            ClearMsalCache();
+            AssertMsalCacheIsEmpty();
+
+            // passing empty password to make sure that token returned silenlty - using RT
+            adalAuthResult = await adal::Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContextIntegratedAuthExtensions.
+                AcquireTokenAsync(adalContext, AdalResource1, ClientId, 
+                new adal::Microsoft.IdentityModel.Clients.ActiveDirectory.UserPasswordCredential(user.Upn, ""));
+
+            ValidateAdalAuthResult();
```

Comment:
suggestion: Clear MSAL cache again and call msal.aquiretoken to ensure msal also uses the adal cache, this way this test will validate that both libraries can use items from just the adal cache created with MSAL

Reply:
Good point , I have missed that case, will add and rename test names to better reflect test purpose